src/lib/y2partitioner/widgets/lvm_validations.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/lvm_vg"
+require "y2storage/lvm_lv"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Widgets
+    # Mixin for validations before start a LVM wizard
+    #
+    # This validations are typically performed in the buttons that launch a wizard.
+    #
+    # @see AddLvmButton
+    module LvmValidations
+      # TODO
+      def validate_add_vg
+        return true
+      end
+
+      # Validates that a new lv might be added to a specific vg
+      #
+      # @note Popup error messages are presented when some validations fail.
+      #
+      # @param vg [LvmVg]
+      # @return [Boolean]
+      def validate_add_lv(vg)
+        if vg.nil?
+          Yast::Popup.Error(_("No device selected"))
+          return false
+        end
+
+        if vg.number_of_free_extents == 0
```

Comment:
I'm not going to do a full review, but I'm curious what's the point/reason to introduce this redundancy with `Sequences::AddLvmLv#preconditions`.

Reply:
Well, the reason is because showing the popup error once the wizard has started results in a weird UI behaviour (I will do a screenshots). To avoid that, it is necessary to validate and show the popup before launching the wizard.